src/Oro/Component/Config/Loader/YamlCumulativeFileLoader.php

```diff
@@ -18,18 +19,32 @@ protected function loadFile($file)
     /**
      * Parses a YAML file.
      *
-     * @param string $file Path to a file
+     * @param string $filePath Path to a file
      * @return array|null
      * @throws \InvalidArgumentException When loading of YAML file returns error
      */
-    protected function parseFile($file)
+    protected function parseFile($filePath)
     {
-        $yamlParser = new YamlParser();
         try {
-            return $yamlParser->parse(file_get_contents($file));
+            $configData = Yaml::parse(file_get_contents($filePath)) ? : [];
+
+            if (array_key_exists('imports', $configData) && is_array($configData['imports'])) {
+                $imports = $configData['imports'];
+                unset($configData['imports']);
+
+                foreach ($imports as $importData) {
+                    if (array_key_exists('resource', $importData)) {
+                        $file = new \SplFileInfo($filePath);
+                        $import = new \SplFileInfo($file->getPath() . DIRECTORY_SEPARATOR . $importData['resource']);
+                        $configData = array_merge_recursive($configData, $this->parseFile($import));
```

Comment:
Usually but not in this case. The maximum nesting level cannot be hardcoded, it is subjective (I MIGHT need 256 includes, or MORE....).

Adding a new argument in the class constructor will do, but it will be invasive, so for now I keep this as it is.